src/NServiceBus.Core/Routing/NativePublishSubscribeFeature.cs

```diff
@@ -8,6 +9,7 @@ public NativePublishSubscribeFeature()
         {
             EnableByDefault();
             Prerequisite(c => c.Settings.Get<TransportInfrastructure>().OutboundRoutingPolicy.Publishes == OutboundRoutingType.Multicast, "The transport does not support native pub sub");
+            Prerequisite(c => HybridSubscriptions.IsMigrationModeEnabled(c.Settings) == false, "The transport has enabled pub sub migration mode");
```

Comment:
should we also add this check to the message driven implementation for completeness sake? This could prevent incorrect cases where someone would enable the hyrid mode with a transport not supporting native pubsub. In that case, it would just "silently" stick with message driven pubsub instead of failing.

Reply:
No, I don't think we need it there. It's the transport's responsibility to expose the migration API, so there's no way for someone to enable it in the scenario you're describing.

When transports add native support, they will be reporting multicast to core, and then turning migration mode as needed.